### PR TITLE
ATO-736: Adds New Signing Key MFA Reset JARs

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
## What:
Adds a new KMS key for signing the JARs for MFA reset

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: ci/terraform/oidc/ecc-signing-key.tf [hunk 1]
```diff
@@ -64,8 +64,10 @@ data "aws_iam_policy_document" "storage_token_signing_key_access_policy" {
   }
 }
 
+//NOTE: This key is used to sign the storage token included as a claim inside the
```
Comment (billrdunn) [inline ci/terraform/oidc/ecc-signing-key.tf:67]: I would try and get the information in this note into the `description` below

File: ci/terraform/oidc/ecc-signing-key.tf [hunk 2]
```diff
@@ -96,3 +98,37 @@ data "aws_iam_policy_document" "mfa_reset_signing_key_access_policy" {
     }
   }
 }
+
+
+resource "aws_kms_key" "mfa_reset_jar_signing_key_ecc" {
+  description              = "KMS signing key (ECC) for MFA reset JARs when resetting MFA"
```
Comment (billrdunn) [inline ci/terraform/oidc/ecc-signing-key.tf:104]: Maybe something like
"KMS signing key (ECC) for JARs sent from Authentication to IPV for MFA reset"

File: ci/terraform/oidc/ecc-signing-key.tf [hunk 3]
```diff
@@ -64,8 +64,10 @@ data "aws_iam_policy_document" "storage_token_signing_key_access_policy" {
   }
 }
 
+//NOTE: This key is used to sign the storage token included as a claim inside the
+//MFA reset JAR.
 resource "aws_kms_key" "mfa_reset_signing_key_ecc" {
```
Comment (billrdunn) [inline ci/terraform/oidc/ecc-signing-key.tf:69]: Could rename this `mfa_reset_token_signing_key_ecc`?

File: ci/terraform/oidc/ecc-signing-key.tf (hunk 2 above)
Comment (Ryan-Andrews99) [reply ci/terraform/oidc/ecc-signing-key.tf:104]: That makes more sense than what I wrote 😄

File: ci/terraform/oidc/ecc-signing-key.tf (hunk 3 above)
Comment (Ryan-Andrews99) [reply ci/terraform/oidc/ecc-signing-key.tf:69]: Did you mean something like `mfa_reset_storage_token_signing_key_ecc` ?

File: ci/terraform/oidc/ecc-signing-key.tf (hunk 1 above)
Comment (Ryan-Andrews99) [reply ci/terraform/oidc/ecc-signing-key.tf:67]: Will update description, re-naming should help make it clearer

File: ci/terraform/oidc/ecc-signing-key.tf (hunk 3 above)
Comment (billrdunn) [reply ci/terraform/oidc/ecc-signing-key.tf:69]: Yep that works too 👍

**Review by billrdunn — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]

**Pull request merged**